src/push/monitor.ts

```diff
@@ -36,7 +36,8 @@ export type MonitorSchema = Omit<MonitorConfig, 'locations'> & {
   filter: Monitor['filter'];
 };
 
-function translateLocation(locations: MonitorConfig['locations']) {
+function translateLocation(locations?: MonitorConfig['locations']) {
+  if (!locations) return [];
```

Comment:
Do we need this? Does privateLocations also expect a default array when not set?

Reply:
i think this can be removed. i was getting some validation error But it was probably unrelated to this.

Reply:
@vigneshshanmugam, actually I looked it over again, and this allows a user to only define private locations if desired. Otherwise, you get a reference error for the map function. This will be critical for off-cloud users.

We can also define a default value instead of the conditional. 

```
function translateLocation(locations: MonitorConfig['locations'] = []) {
  return locations.map(loc => LocationsMap[loc]).filter(Boolean);
}
```
@vigneshshanmugam let me know your preferences.

Reply:
Gotcha, thanks for the details. I am fine with both. Let's get this in.